test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs

```diff
@@ -153,6 +154,62 @@ public static void FileDoesNotExist(MSBuildResult result, params string[] paths)
             }
         }
 
+        public static void NuspecContains(MSBuildResult result, string nuspecPath, string expected)
+        {
+            if (result == null)
+            {
+                throw new ArgumentNullException(nameof(result));
+            }
+
+            if (nuspecPath == null)
+            {
+                throw new ArgumentNullException(nameof(nuspecPath));
+            }
+
+            if (expected == null)
+            {
+                throw new ArgumentNullException(nameof(expected));
+            }
+
+            nuspecPath = Path.Combine(result.Project.DirectoryPath, nuspecPath);
+            FileExists(result, nuspecPath);
+
+            var content = File.ReadAllText(nuspecPath);
+            if (!content.Contains(expected))
+            {
+                throw new NuspecException(result, nuspecPath, content, expected);
+            }
+        }
+
+        public static void NupkgContains(MSBuildResult result, string nupkgPath, string filePath)
+        {
+            if (result == null)
+            {
+                throw new ArgumentNullException(nameof(result));
+            }
+
+            if (nupkgPath == null)
+            {
+                throw new ArgumentNullException(nameof(nupkgPath));
+            }
+
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            nupkgPath = Path.Combine(result.Project.DirectoryPath, nupkgPath);
+            FileExists(result, nupkgPath);
+
+            var unzipped = Path.Combine(result.Project.DirectoryPath, "nupkg");
+            ZipFile.ExtractToDirectory(nupkgPath, unzipped);
```

Comment:
You could look at `ZipFile.Entries` instead of extracting it. This is fine too.

Reply:
I considered that, but kept the extract for debugging. It's nice to be able to break in the test at the point of failure and then just step through and open the folder